runtime/src/main/cpp/Memory.cpp

```diff
@@ -87,11 +89,19 @@ constexpr double kGcToComputeRatioThreshold = 0.5;
 // Never exceed this value when increasing GC threshold.
 constexpr size_t kMaxErgonomicThreshold = 32 * 1024;
 // Threshold of size for toFree set, triggering actual cycle collector.
-constexpr size_t kMaxToFreeSize = 8 * 1024;
+constexpr size_t kMaxToFreeSizeThreshold = 8 * 1024;
+// Never exceed this value when increasing size for toFree set, triggering actual cycle collector.
+constexpr size_t kMaxErgonomicToFreeSizeThreshold = 8 * 1024 * 1024;
 // How many elements in finalizer queue allowed before cleaning it up.
 constexpr size_t kFinalizerQueueThreshold = 32;
 // If allocated that much memory since last GC - force new GC.
 constexpr size_t kMaxGcAllocThreshold = 8 * 1024 * 1024;
+// If GC base work to collection cycles time ratio is greater this value,
```

Comment:
> If GC base work to collection cycles time ratio

Seems wrong.

Reply:
// If the ratio of GC collection cycles time to program execution time is greater this value,
// increase GC threshold for cycles collection.

Reply:
lgtm.